source/templates.tex

```diff
@@ -367,28 +367,31 @@
 A non-type \grammarterm{template-parameter}
 shall have one of the following (optionally cv-qualified) types:
 \begin{itemize}
-\item a literal type that
-  has strong structural equality\iref{class.compare.default},
-
-\item an lvalue reference type,
-
+\item a structural type (see below),
 \item a type that contains a placeholder type\iref{dcl.spec.auto}, or
-
 \item a placeholder for a deduced class type\iref{dcl.type.class.deduct}.
 \end{itemize}
-
-\pnum
-\begin{note}
-Other types are disallowed either explicitly below or implicitly by
-the rules governing the form of
-\grammarterm{template-argument}{s}\iref{temp.arg}.
-\end{note}
 The top-level
 \grammarterm{cv-qualifier}{s}
 on the
 \grammarterm{template-parameter}
 are ignored when determining its type.
 
+\pnum
+A \defnadj{structural}{type} is one of the following:
+\begin{itemize}
+\item a scalar type, or
+\item an lvalue reference type, or
+\item a literal class type with the following properties:
+\begin{itemize}
+\item
+all base classes and non-static data members are public and non-mutable and
+\item
+the types of all bases classes and non-static data members are
+structural types or (possibly multi-dimensional) array thereof.
+\end{itemize}
+\end{itemize}
```

Comment:
I think this is missing some indentation according to https://github.com/cplusplus/draft/wiki/Specification-Style-Guidelines#latex-source-code-formatting.

Reply:
We're not yet sure what to do with `\item`; see #3127. Thus, the formatting used here is as good as (most) any.